File: scripts/convert-heic.js

const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

const inputDir = path.join(__dirname, '../../images');
const outputDir = path.join(__dirname, '../public/img/extern');

if (!fs.existsSync(outputDir)) {
    fs.mkdirSync(outputDir, { recursive: true });
}

fs.readdir(inputDir, (err, files) => {
    if (err) {
        console.error('Could not list the directory.', err);
        process.exit(1);
    }

    files.forEach((file, index) => {
        const ext = path.extname(file).toLowerCase();
        if (ext === '.heic') {
            const inputFile = path.join(inputDir, file);
            const outputFile = path.join(outputDir, path.basename(file, path.extname(file)) + '.jpg');

            sharp(inputFile)
                .jpeg({ quality: 90 })
                .toFile(outputFile)
                .then(() => {
                    console.log(`Converted: ${file} -> ${path.basename(outputFile)}`);
                })
                .catch(err => {
                    console.error(`Error converting ${file}:`, err);
                });
        }
    });
});

File: scripts/convert_heic.py

import os
from PIL import Image
from pillow_heif import register_heif_opener

register_heif_opener()

input_dir = os.path.join(os.path.dirname(__file__), '../../images')
output_dir = os.path.join(os.path.dirname(__file__), '../public/img/extern')

if not os.path.exists(output_dir):
    os.makedirs(output_dir)

print(f"Scanning directory: {input_dir}")

for filename in os.listdir(input_dir):
    if filename.lower().endswith('.heic'):
        input_path = os.path.join(input_dir, filename)
        output_filename = os.path.splitext(filename)[0] + '.jpg'
        output_path = os.path.join(output_dir, output_filename)

        try:
            image = Image.open(input_path)
            image.convert('RGB').save(output_path, "JPEG", quality=90)
            print(f"Converted: {filename} -> {output_filename}")
        except Exception as e:
            print(f"Failed to convert {filename}: {e}")
